twilight-mention/src/fmt.rs

```diff
@@ -189,13 +282,45 @@ impl Mention<Id<UserMarker>> for User {
     }
 }
 
+/// Components to construct a slash command mention.
+///
+/// Format slash commands, subcommands and subcommand groups.
+/// See [Discord Docs/Message Formatting].
+/// See [Discord Docs Changelog/Slash Command Mentions].
+///
+/// [Discord Docs/Message Formatting]: https://discord.com/developers/docs/reference#message-formatting
+/// [Discord Docs Changelog/Slash Command Mentions]: https://discord.com/developers/docs/change-log#slash-command-mentions
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum CommandMention {
+    // Field order is the same as the mention format.
+    Command {
```

Comment:
turning the tuples into structs, these should wrap over the structs, or the implementations could be removed and users could rely on this enum entirely

Reply:
removed tuples, more structs would be bloat